src/lib/storage/zero_suppression/encode_by_zs_type.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <cstdint>
+#include <memory>
+
+#include "base_zero_suppression_vector.hpp"
+#include "zs_type.hpp"
+#include "zs_vector_meta_info.hpp"
+
+#include "types.hpp"
+
+namespace opossum {
+
+/**
+ * @brief Encodes a vector of uint32_t using a given ZsType
+ *
+ * @param meta_info optional struct that provides the encoding algorithms with additional information
+ */
+std::unique_ptr<BaseZeroSuppressionVector> encode_by_zs_type(ZsType type, const pmr_vector<uint32_t>& vector,
```

Comment:
Is this the main interface to the whole zero suppression thing? If so, it is well hidden and I would consider moving it to `storage/`

Reply:
I wouldn’t. It’s not something that a lot of people need to use.

Reply:
Then what is? If I have a vector and want to compress it, what is the main interface?